src/modules/Hosts/HostsUILib/Helpers/HostsService.cs

```diff
@@ -197,7 +198,16 @@ public async Task WriteAsync(string additionalLines, IEnumerable<Entry> entries)
                     _backupDone = true;
                 }
 
-                await _fileSystem.File.WriteAllLinesAsync(HostsFilePath, lines, Encoding);
+                // FileMode.OpenOrCreate is necessary to prevent UnauthorizedAccessException when the hosts file is hidden
+                using var stream = _fileSystem.FileStream.Create(HostsFilePath, FileMode.OpenOrCreate, FileAccess.Write, FileShare.Read, _defaultBufferSize, FileOptions.Asynchronous);
+                using var writer = new StreamWriter(stream, Encoding);
+                foreach (var line in lines)
+                {
+                    await writer.WriteLineAsync(line.AsMemory());
```

Comment:
Just checking.. there's no way to write entire file at once with StreamWriter?

Reply:
I don't think there is one, I did a double check and can't find anything.
AFAIK usually the wrapper to do this at once is `File.WriteAllLinesAsync(...)`.
Alternative could be `await writer.WriteLineAsync(string.Join(Environment.NewLine, lines))` but personally don't like it and I expect some overread.

Reply:
Ok. As I said. Just checking.. :)